### PR TITLE
Elasticsearch: Replace error source http client with a new error source methods

### PR DESCRIPTION
**Pull request by ivanahuckova**
We want to deprecate and get rid of `errorsource/httpclient` as a part of https://github.com/grafana/grafana-plugin-sdk-go/issues/1105, as error source middleware implemented in experimental error source httpclient is turning any non-200 response to error and hiding original message. 

 Instead of that we have introduced 2 methods developers can use:
 - `backend.ErrorSourceFromHTTPStatus` to get error source from http code
 - `backend.IsDownstreamHTTPError` to get error source from http error

In this PR we are removing `errorsource/httpclient` and instead just using those methods.

### CODE REVIEW

File: pkg/tsdb/elasticsearch/data_query.go [hunk 1]
```diff
@@ -77,10 +77,17 @@ func (e *elasticsearchDataQuery) execute() (*backend.QueryDataResponse, error) {
...
 	res, err := e.client.ExecuteMultisearch(req)
 	if err != nil {
-		// We are returning error containing the source that was added through errorsource.Middleware
+		if backend.IsDownstreamHTTPError(err) {
+			return errorsource.AddErrorToResponse(e.dataQueries[0].RefID, response, errorsource.DownstreamError(err, false)), nil
+		}
 		return errorsource.AddErrorToResponse(e.dataQueries[0].RefID, response, err), nil
 	}
 
+	if res.Status/100 != 2 {
```
Comment (ivanahuckova) [inline pkg/tsdb/elasticsearch/data_query.go]: Same as https://github.com/grafana/grafana-plugin-sdk-go/blob/main/experimental/errorsource/error_source_middleware.go#L21

File: pkg/tsdb/elasticsearch/error_handling_test.go [hunk 2]
```diff
@@ -43,7 +43,7 @@ func TestErrorAvgMissingField(t *testing.T) {
 	require.NoError(t, err)
 
 	// FIXME: we should return the received error message
-	require.Len(t, result.response.Responses, 0)
+	require.Equal(t, "unexpected status code: 400", result.response.Responses["A"].Error.Error())
```
Comment (ivanahuckova) [inline pkg/tsdb/elasticsearch/error_handling_test.go:46]: Previously, the middleware returned error as an error, see https://github.com/grafana/grafana-plugin-sdk-go/blob/main/experimental/errorsource/error_source_middleware.go#L26. We are now moving processing to Elasticsearch and returning it as error response.

File: pkg/tsdb/elasticsearch/data_query.go [hunk 3]
```diff
@@ -77,10 +77,17 @@ func (e *elasticsearchDataQuery) execute() (*backend.QueryDataResponse, error) {
 	e.logger.Info("Prepared request", "queriesLength", len(queries), "duration", time.Since(start), "stage", es.StagePrepareRequest)
 	res, err := e.client.ExecuteMultisearch(req)
 	if err != nil {
-		// We are returning error containing the source that was added through errorsource.Middleware
+		if backend.IsDownstreamHTTPError(err) {
+			return errorsource.AddErrorToResponse(e.dataQueries[0].RefID, response, errorsource.DownstreamError(err, false)), nil
+		}
```
Comment (njvrzm) [inline pkg/tsdb/elasticsearch/data_query.go:82]: Maybe slightly cleaner here to say 
```suggestion
		if backend.IsDownstreamHTTPError(err) {
			err = errorsource.DownstreamError(err, false)
		}
```

**Review by njvrzm — APPROVED** [review]
Awesome! Thanks for this. One minor and optional suggestion.

File: pkg/tsdb/elasticsearch/data_query.go (hunk 3 above)
Comment (ivanahuckova) [reply pkg/tsdb/elasticsearch/data_query.go]: Great suggestions! Thanks. Applied. 🙂

**Pull request merged**